dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -165,11 +166,22 @@ export const FrontLayout = ({ front, NAV }: Props) => {
 							editionId={front.editionId}
 						>
 							<DecideContainer
+								collectionId={collection.id}
 								trails={trails}
 								containerType={collection.collectionType}
 								containerPalette={collection.containerPalette}
 								showAge={collection.displayName === 'Headlines'}
+								hasMore={collection.hasMore}
 							/>
+							{collection.hasMore && (
+								<Island deferUntil={'visible'}>
+									<ShowMore
+										pageId={front.pressedPage.id}
+										collectionId={collection.id}
+										displayName={collection.displayName}
+									/>
+								</Island>
+							)}
```

Comment:
Does this mean that each container on the page will download, evaluate and execute the show more island?

Reply:
As discussed offline, we should look at having a central island, one that only gets downloaded and executed once, which acts as the controller for these buttons.

We could use some plain javascript here that loops all the placeholders and sets an event listener on each to fetch and render the content

Reply:
I think @AshCorr 's [PR had a similar issue](https://github.com/guardian/dotcom-rendering/pull/4905/files) as this which suggests we have a design constraint and probably don't want to ever have an island inside a container.

Your use case is more complex in that you're making a fetch but hopefully there will be useful similarities.